src/deepsparse/v2/pipeline.py

```diff
@@ -13,90 +13,119 @@
-
-    _scheduler_group: SchedulerGroup = PrivateAttr()
-
-    class Config:
-        arbitrary_types_allowed = True
-
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def __init__(
+        self,
+        ops: Union[Dict[str, Operator], List[Operator]],
+        router: Router,
+        schedulers: List[OperatorScheduler],
+    ):
 
+        self.ops = ops
+        self.router = router
+        self.schedulers = schedulers
         self.validate()
 
         # SchedulerGroup handles running all schedulers in order of priority
         self._scheduler_group = SchedulerGroup(self.schedulers)
 
+    def run(self, inp: Any, context: Optional[Context]):
+        """
+        Run through the operators using the provided router and scheduler. Update the
+        context to reflect each step of the router. The input to a given operator is the
+        output of the previous operator.
+
+        :param inp: input to the operator. expected to be of any type that is
+        expected by the operator.
+        :param context: context to store the current the inputs, outputs, and operator
+        for each step of the router.
+
+        """
+        next_step = self.router.START_ROUTE
+        while next_step != self.router.END_ROUTE:
+            # Either a dictionary key or valid index
+            operator = self.ops[next_step]
+
+            output_future = self._scheduler_group.submit(
+                operator=operator, operator_input=inp, context=context
+            )
+
+            # wait for future to resolve
+            operator_output = output_future.result()
+
+            # update context
+            context.update(
+                operator=operator,
+                input=inp,
+                output=operator_output,
+            )
+
+            next_step = self.router.next(next_step, self.ops)
+            inp = operator_output
+        return operator_output, context
+
     def __call__(self, *args, return_context: bool = False, **kwargs):
         """
-        :param return_context: if True, retrns tuple of the pipelien output
+        :param return_context: if True, returns tuple of the pipeline output
             and entire context. Default False
-        :return: output of the pipeline stages ran with the router for the given input
+        :return: output of the pipeline operators ran with the router for the given
+        input
         """
         if len(args) > 1:
             raise ValueError(
-                "Only 1 in-line argument may be supplied to Pipeline which "
-                f"must be a Schema, found: {len(args)}"
+                "Only 1 unnamed arg may be supplied to a Pipeline, the input expected"
+                "for the first Operator."
             )
         if args and kwargs:
             raise ValueError(
-                "Pipeline can only run either a single in-line argument schema or a "
+                "Pipeline can only run either a in-line arguments or a "
```

Comment:
This isn't new code, but I think the inclusion of this exception violates the [Liskov Substitution Principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle) since it prevents `Pipeline` from being able to act as an `Operator`.  This is because [`Scheduler.submit` explicitly calls a given operator with both an argument and a keyword argument](https://github.com/neuralmagic/deepsparse/blob/e5220256ccbf320df33af4a8c32d61128ceed729/src/deepsparse/v2/schedulers/scheduler.py#L55). YMMV, but @horheynm and I ran into some sadness because of this 😭

Reply:
I see what you're saying but I guess we've kinda reserved args/kwargs for inputs which may change between operators/pipelines and made them responsible for resolving the appropriate schema needed. So you could apply the condition above to the operator as well and it would be fine. The other positional arguments (`return_context` in pipeline and `context` in operators) are passed in explicitly.

Reply:
Ran through an example from @horheynm and I see where this is causing a problem. Thanks!